src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataFilter.cs

```diff
@@ -49,15 +50,17 @@ public void OnResourceExecuting(ResourceExecutingContext context)
                     SaveTempData(
                         result: null,
                         factory: saveTempDataContext.TempDataDictionaryFactory,
+                        filters: saveTempDataContext.Filters,
```

Comment:
The original approach had a bug that we fixed during 1.1.0. The only way to avoid the ordering problem that arises, is to make this part of the `SaveTempDataFilter`. I did the minimal change here to keep from breaking anything